service/history/replication/stream_sender.go

```diff
@@ -107,8 +116,19 @@ func (s *StreamSenderImpl) Start() {
 	) {
 		return
 	}
+	getSenderEventLoop := func(priority enumsspb.TaskPriority) func() error {
+		return func() error {
+			return s.sendEventLoop(priority)
+		}
+	}
+
+	if s.isTieredStackEnabled {
```

Comment:
this requires restart when dc is updated. could it just use config.EnableReplicationTaskTieredProcessing()?